app/controllers/two_factor_authentication/add_methods_controller.rb

```diff
@@ -0,0 +1,43 @@
+module TwoFactorAuthentication
+  class AddMethodsController < ApplicationController
+    before_action :confirm_two_factor_authenticated
+
+    def index
+      @two_factor_add_methods_form = TwoFactorAddMethodsForm.new
+      @presenter = two_factor_options_presenter(after_sign_in_path_for(current_user))
+    end
+
+    def create
+      @two_factor_add_methods_form = TwoFactorAddMethodsForm.new
+      result = @two_factor_add_methods_form.submit(two_factor_add_methods_form_params)
+
+      if result.success?
+        process_valid_form
+      else
+        @presenter = two_factor_options_presenter
+        render :index
+      end
+    end
+
+    private
+
+    def two_factor_options_presenter(next_step_url = nil)
+      TwoFactorAddMethodsPresenter.new(current_user, current_sp, view_context, next_step_url)
+    end
+
+    def process_valid_form
+      case @two_factor_add_methods_form.selection
+      when 'sms', 'voice'
+        redirect_to phone_setup_url
+      when 'auth_app'
+        redirect_to authenticator_setup_url
+      when 'piv_cac'
+        redirect_to setup_piv_cac_url
+      end
+    end
```

Comment:
What do you think about creating a mapping of 2FA options to endpoints to eliminate this case statement and prevent it from growing as we add more options, something like:
```ruby
module TwoFactorOptions
  ENDPOINTS = {
    sms: 'phone_setup_url',
    voice: 'phone_setup_url',
    auth_app: 'authenticator_setup_url',
    piv_cac: 'setup_piv_cac_url'
  }
end
```
and in the controller:
```ruby
def process_valid_form
  selection = @two_factor_add_methods_form.selection
  url = TwoFactorOptions::ENDPOINTS.fetch(selection, 'phone_setup_url')
  redirect_to public_send(url)
end
```

Reply:
Have something similar in the other controller.